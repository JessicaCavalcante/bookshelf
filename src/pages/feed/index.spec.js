import { Feed } from './index.js';
import * as post from '../../components/post/index.js';
import * as services from '../../services/index.js';

describe('Feed', () => {
  it('should be a function', () => {
    expect(typeof Feed).toBe('function');
  });

  it('should load the page', () => {
    services.getReviews = jest.fn(() => Promise.resolve(true));
    expect(Feed()).toMatchSnapshot();
  });

  it('should have return the header html when getReviews is empty', () => {
    services.getReviews = jest.fn(() => Promise.resolve([]));
    const result = Feed().innerHTML;
    expect(result).toBe('<header><h1 class="title">Livros</h1></header>');
  });

  it('should be called getReviews with true parameter', () => {
    services.getReviews = jest.fn(() => Promise.resolve([]));
    Feed();
    expect(services.getReviews).toHaveBeenCalledWith(true);
  });

  it('should be  called Post after called Feed and getReviews have true parameter', () => {
    services.getReviews = jest.fn(() => Promise.resolve([{}]));
    post.Post = jest.fn(() => document.createElement('div'));
    Feed();
    expect(services.getReviews).toHaveBeenCalledWith(true);
  });
});
