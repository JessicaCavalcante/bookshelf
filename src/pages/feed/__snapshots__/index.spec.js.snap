// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed should load the page 1`] = `
<div>
  <header>
    <h1
      class="title"
    >
      Livros
    </h1>
  </header>
</div>
`;
